packages/ogmios/src/CardanoNode/OgmiosCardanoNode.ts

```diff
@@ -1,96 +1,57 @@
+import { BehaviorSubject, firstValueFrom, takeUntil } from 'rxjs';
 import {
-  Cardano,
   CardanoNode,
   EraSummary,
   GeneralCardanoNodeError,
   GeneralCardanoNodeErrorCode,
   HealthCheckResponse,
   StakeDistribution
 } from '@cardano-sdk/core';
-import {
-  ConnectionConfig,
-  LedgerStateQuery,
-  createConnectionObject,
-  createLedgerStateQueryClient,
-  getServerHealth
-} from '@cardano-ogmios/client';
 import { Logger } from 'ts-log';
-import { RunnableModule, contextLogger } from '@cardano-sdk/util';
-import { createInteractionContextWithLogger, ogmiosServerHealthToHealthCheckResponse } from '../util';
-import { queryEraSummaries, withCoreCardanoNodeError } from './queries';
+import { OgmiosObservableCardanoNode } from './OgmiosObservableCardanoNode';
+import { RunnableModule } from '@cardano-sdk/util';
 
 /**
  * Access cardano-node APIs via Ogmios
  *
  * @class OgmiosCardanoNode
  */
 export class OgmiosCardanoNode extends RunnableModule implements CardanoNode {
-  #stateQueryClient: LedgerStateQuery.LedgerStateQueryClient;
-  #logger: Logger;
-  #connectionConfig: ConnectionConfig;
+  #ogmiosObservableCardanoNode: OgmiosObservableCardanoNode;
+  #cancelAll$: BehaviorSubject<void>;
 
-  constructor(connectionConfig: ConnectionConfig, logger: Logger) {
+  constructor(ogmiosObservableCardanoNode: OgmiosObservableCardanoNode, logger: Logger) {
     super('OgmiosCardanoNode', logger);
-    this.#logger = contextLogger(logger, 'OgmiosCardanoNode');
-    this.#connectionConfig = connectionConfig;
+    this.#ogmiosObservableCardanoNode = ogmiosObservableCardanoNode;
+    this.#cancelAll$ = new BehaviorSubject<void>(undefined);
   }
 
   public async initializeImpl(): Promise<void> {
-    this.#logger.info('Initializing CardanoNode');
-    this.#stateQueryClient = await createLedgerStateQueryClient(
-      await createInteractionContextWithLogger(this.#logger, { connection: this.#connectionConfig })
-    );
-    this.#logger.info('CardanoNode initialized');
+    return Promise.resolve();
   }
 
   public async shutdownImpl(): Promise<void> {
-    this.#logger.info('Shutting down CardanoNode');
-    await this.#stateQueryClient.shutdown();
+    this.#cancelAll$.next();
+    return Promise.resolve();
   }
 
   public async eraSummaries(): Promise<EraSummary[]> {
     this.#assertIsRunning();
-    return queryEraSummaries(this.#stateQueryClient, this.#logger);
+    return firstValueFrom(this.#ogmiosObservableCardanoNode.eraSummaries$.pipe(takeUntil(this.#cancelAll$)));
```

Comment:
Best to map the error to one of the CardanoNode errors, as `firstValueFrom` [will throw an EmptyError if the observable completes without emitting](https://rxjs.dev/api/index/function/firstValueFrom#description)
```suggestion
    return firstValueFrom(this.#ogmiosObservableCardanoNode.eraSummaries$.pipe(takeUntil(this.#cancelAll$))).pipe(catchError(error => {
    throw new CardanoNodeError(...)
    }));
```